x-pack/plugins/ml/server/models/data_frame_analytics/validation.ts

```diff
@@ -0,0 +1,299 @@
+  'xpack.ml.models.dfaValidation.messages.dependentVarHeading',
+  {
+    defaultMessage: 'Dependent variable',
+  }
+);
+const dependentVarWarningMessage = {
+  id: 'dep_var_check',
+  text: '',
+  status: VALIDATION_STATUS.WARNING,
+  heading: dependentVarHeading,
+};
+const analysisFieldsWarningMessage = {
+  id: 'analysis_fields',
+  text: '',
+  status: VALIDATION_STATUS.WARNING,
+  heading: analysisFieldsHeading,
+};
+
+function getTrainingPercentMessage(trainingDocs: number) {
+  let trainingPercentMessage;
+
+  if (trainingDocs >= TRAINING_DOCS_UPPER) {
+    trainingPercentMessage = {
+      id: 'training_percent_high',
+      text: i18n.translate('xpack.ml.models.dfaValidation.messages.highTrainingPercentWarning', {
+        defaultMessage:
+          'A high number of training docs may result in long running jobs. Try reducing the training percent.',
+      }),
+      status: VALIDATION_STATUS.WARNING,
+      heading: trainingPercentHeading,
+    };
+  } else if (trainingDocs <= TRAINING_DOCS_LOWER) {
+    trainingPercentMessage = {
+      id: 'training_percent_low',
+      text: i18n.translate('xpack.ml.models.dfaValidation.messages.lowTrainingPercentWarning', {
+        defaultMessage:
+          'A low number of training docs may result in inaccurate models. Try increasing the training percent or using a larger data set.',
+      }),
+      status: VALIDATION_STATUS.WARNING,
+      heading: trainingPercentHeading,
+    };
+  } else {
+    trainingPercentMessage = {
+      id: 'training_percent',
+      text: i18n.translate('xpack.ml.models.dfaValidation.messages.trainingPercentSuccess', {
+        defaultMessage: 'The training percent is high enough to model patterns in the data.',
+      }),
+      status: VALIDATION_STATUS.SUCCESS,
+      heading: trainingPercentHeading,
+    };
+  }
+
+  return trainingPercentMessage;
+}
+
+async function getValidationCheckMessages(
+  asCurrentUser: IScopedClusterClient['asCurrentUser'],
+  analyzedFields: string[],
+  index: string | string[],
+  query: any = defaultQuery,
+  depVar: string,
+  trainingPercent?: number
+) {
+  const messages = [];
+  const emptyFields: string[] = [];
+  const percentEmptyLimit = FRACTION_EMPTY_LIMIT * 100;
+  let depVarValid = true;
+  let analysisFieldsNumHigh = false;
+  let analysisFieldsEmpty = false;
+
+  const fieldLimit =
+    analyzedFields.length <= MINIMUM_NUM_FIELD_FOR_CHECK
+      ? analyzedFields.length
+      : MINIMUM_NUM_FIELD_FOR_CHECK;
+
+  let aggs = analyzedFields.slice(0, fieldLimit).reduce((acc, curr) => {
+    acc[curr] = { missing: { field: curr } };
+    return acc;
+  }, {} as any);
+
+  if (depVar !== '') {
+    const depVarAgg =
+      depVar !== ''
+        ? {
+            [`${depVar}_const`]: {
+              cardinality: { field: depVar },
+            },
+          }
+        : {};
+
+    aggs = { ...aggs, ...depVarAgg };
+  }
+
+  try {
+    const { body }: { body: ValidationSearchResult } = await asCurrentUser.search({
```

Comment:
you should be able to use the generic on `search` here.
something like
```
const { body } = await asCurrentUser.search<ValidationSearchResult>(

```

Reply:
Updated in a7e5fc69c41f0b104667516fc53457eebf0a649d